self_test/SelfTest.ts

```diff
@@ -0,0 +1,120 @@
+
+import { Injector } from 'reduct'
+import Config from '../src/services/Config'
+import { create as createLogger } from '../src/common/log'
+const ilpFetch = require('ilp-fetch')
+const log = createLogger('SelfTest')
+const WebSocket = require('ws')
+const manifestJson = require('./self-test-manifest.json')
+const axios = require('axios')
+const crypto = require('crypto')
+export default class SelfTest {
+  private config: Config
+  constructor (deps: Injector) {
+    this.config = deps(Config)
+  }
+
+  start () {
+    if (!this.config.disableSelfTest) {
+      this.run()
+      .catch(err => log.error(err))
+    } else {
+      log.debug('Skipping self test')
+    }
+  }
+
+  async run () {
+    const duration = 300
+    const host = this.config.publicUri
+    try {
+      const randomName = crypto.randomBytes(20).toString('hex')
+      manifestJson['manifest']['name'] = randomName
+      log.debug('manifestJson', manifestJson)
+      let response = await ilpFetch(`${host}/pods?duration=${duration}`, {
+        headers: {
+          Accept: `application/codius-v1+json`,
+          'Content-Type': 'application/json'
+        },
+        maxPrice: (this.config.hostCostPerMonth * 1000000).toString(),
+        method: 'POST',
+        body: JSON.stringify(manifestJson),
+        timeout: 70000 // 1m10s
+      })
+      log.debug('ilpFetch Resp', response)
```

Comment:
Logging a response object can be annoying in the logs because it has a lot of methods on it. It might be better to just print the statusCode instead, or at least use the trace level if you want to dump that much information